src/autonat/index.ts

```diff
@@ -132,20 +132,20 @@ export class AutonatService implements Startable {
    * Handle an incoming autonat request
    */
   async handleIncomingAutonatStream (data: IncomingStreamData): Promise<void> {
-    const controller = new TimeoutController(this._init.timeout)
+    const signal = anySignal([AbortSignal.timeout(this._init.timeout)])
```

Comment:
Use of anySignal needs to call `.clear` on the returned `ClearableSignal` otherwise if the passed in `options.signal` is long lived (e.g. one that's aborted when the node is stopped) it can cause memory leaks.

See:

- https://github.com/jacobheun/any-signal/pull/24
- https://github.com/jacobheun/any-signal/issues/23
- https://github.com/jacobheun/any-signal#clearablesignalclear